### PR TITLE
Update Core Package dependencies to support Azsec Pack

### PR DESCRIPTION
**Pull request by henryli001**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Add required dependencies to Mariner Core image to support Azsec Pack

###### Change Log  

- Add 'gawk', 'initscripts' and 'dos2unix' to the base image.

###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/...

###### Test Methodology

- Local Build

### CODE REVIEW

File: SPECS/core-packages/core-packages.spec [hunk 1]
```diff
@@ -84,6 +84,9 @@ Requires:   vim
 Requires:   wget
 Requires:   xz
 Requires:   zlib
+Requires:   gawk
```
Comment (schmittjoseph) [inline SPECS/core-packages/core-packages.spec:87]: Do these packages need to be added to every image, or just the CBL-Mariner Core image? This meta package is used by almost all image configurations, even containers. 


If these packages are targetting just CBL-Mariner Core then the relevant package lists / configs under `toolkit/imageconfigs` should be updated instead.

**Review by schmittjoseph — CHANGES_REQUESTED** [review]

File: SPECS/core-packages/core-packages.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/core-packages/core-packages.spec]: +1. Why cannot we modify the spec for Azsec Pack instead? That sounds like a more appropriate place for dependencies.

Comment (henryli001) [reply SPECS/core-packages/core-packages.spec]: The previous PR title was misleading. Azsec Pack won't be directly enabled in Mariner Core at the moment, but it's needed in images like AFD, AKS, PilotFish etc. But we want to verify azsec pack in Mariner Core to make sure that it works in all derivatives and thus I'm adding the required dependencies here.

**Comment by henryli001** [comment]
No longer needed.

**Pull request closed without merging**